storage/client_test.go

```diff
@@ -1506,6 +1506,226 @@ func TestRetryReadStallEmulated(t *testing.T) {
 	}
 }
 
+// In resumable uploads, the first chunk encounter a stall. Because ChunkTransferTimeout is set to 0,
+// the upload must wait for the entire stall duration.
+func TestRetryWriteReqStallWithDefaultChunkTransferTimeoutEmulated(t *testing.T) {
+	transportClientTest(skipGRPC("service is not implemented"), t, func(t *testing.T, ctx context.Context, project, bucket string, client storageClient) {
+		_, err := client.CreateBucket(ctx, project, bucket, &BucketAttrs{}, nil)
+		if err != nil {
+			t.Fatalf("creating bucket: %v", err)
+		}
+
+		chunkSize := 2 * 1024 * 1024 // 2 MiB
+		fileSize := 5 * 1024 * 1024  // 5 MiB
+		tests := []struct {
+			name         string
+			instructions map[string][]string
+			wantDuration time.Duration
+		}{
+			{
+				name: "stall-on-first-chunk-with-chunk-transfer-timeout-zero",
+				instructions: map[string][]string{
+					"storage.objects.insert": {"stall-for-10s-after-1024K"},
+				},
+				wantDuration: 10 * time.Second,
```

Comment:
So it looks like this test will take >30s to run, which is too much for an emulator test IMO.

Could you maybe rewrite to have the stalls last for 1s, and set a context timeout of 500ms or less on the request? We would expect the requests to fail in this case.

Reply:
Done